core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -272,6 +274,18 @@ object TopicCommand extends Logging {
       val allConfigs = adminClient.describeConfigs(topics.map(new ConfigResource(Type.TOPIC, _)).asJavaCollection).values()
       val liveBrokers = adminClient.describeCluster().nodes().get().asScala.map(_.id())
       val topicDescriptions = adminClient.describeTopics(topics.asJavaCollection).all().get().values().asScala
+      val topicPartitions = topicDescriptions.flatMap(pi => pi.partitions().asScala.map(tpi => new TopicPartition(pi.name(), tpi.partition())))
```

Comment:
I think it would be better to select the partitions to query using the topic command options. In other words, if the user is describing all topics, then we can request all reassignments directly instead of building a big list reassignment request. The common case is that there would be no reassignments in progress.

By the way, the fact that we need two separate APIs means we are sort of doomed to have race conditions in which we get inconsistent results. Probably nothing we can do about it at the moment.

Reply:
Ok, it's a good idea. I'll refactor it so that it first looks at what did the user passed and if it's empty then just initiate an empty list reassignment.